### PR TITLE
Security Libraries 

### PR DESCRIPTION
**Pull request by sushi2k**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue https://github.com/OWASP/owasp-mstg/issues/1068.

### CODE REVIEW

File: Document/0x05b-Basic-Security_Testing.md [hunk 1]
```diff
@@ -440,12 +440,18 @@ rdr pass inet proto tcp from any to any port 5236 -> 127.0.0.1 port 8080
...
+The interception proxy must listen to the port specified in the port forwarding rule above (port 8080).
 
 Start the app and trigger a function that uses FCM. You should see HTTP messages in your interception proxy.
 
 ![Intercepted Messages](Images/Chapters/0x05b/FCM_Intercept.png)
 
+**End-to-End Encryption for Push Notifications**
+
+As an additional layer of security, push notifications can be encrypted by using [Capillary](https://github.com/google/capillary "Capillary"). Capillary is a library to simplify the sending of end-to-end (E2E) encrypted push messages from Java-based application servers to Android clients. 
+
```
Comment (commjoen) [inline Document/0x05b-Basic-Security_Testing.md]: Can you please add references at the end?

File: Document/0x05d-Testing-Data-Storage.md [hunk 2]
```diff
@@ -273,6 +273,16 @@ Inspect the source code to determine whether native Android mechanisms identify
...
 - Use the `store(OutputStream stream, char[] password)` function to store the KeyStore to disk with a password. Make sure that the password is provided by the user, not hard-coded.
 
+##### 3rd Party libraries
+
+There are several different open-source libraries that offer encryption capabilities specific for the Android platform.
+
+- **[Java AES Crypto](https://github.com/tozny/java-aes-crypto "Java AES Crypto")** - A simple Android class for encrypting and decrypting strings.
+- **[SQL Cipher](https://www.zetetic.net/sqlcipher/sqlcipher-for-android/ "SQL Cipher")** - SQLCipher is an open source extension to SQLite that provides transparent 256-bit AES encryption of database files.
+- **[Secure Preferences](https://github.com/scottyab/secure-preferences "Secure Preferences")** - Android Shared preference wrapper than encrypts the keys and values of Shared Preferences.
+
```
Comment (commjoen) [inline Document/0x05d-Testing-Data-Storage.md]: Can you please add references at the end?

File: Document/0x05b-Basic-Security_Testing.md (hunk 1 above)
Comment (sushi2k) [reply Document/0x05b-Basic-Security_Testing.md]: For this section we always use the links as part of the text and didn't have a reference table at the end. I can create an additional table at the end with tools used in this section. Makes also sense for the printed book to see the URLs.

Comment (commjoen) [reply Document/0x05b-Basic-Security_Testing.md]: Maybe it is best to have links here and references as well?

Comment (sushi2k) [reply Document/0x05b-Basic-Security_Testing.md]: Done :-)

**Pull request merged**